plugin/loader/initializer.go

```diff
@@ -19,8 +20,9 @@ func initialize(plugins []plugin.Plugin) error {
 	return nil
 }
 
-func run(plugins []plugin.Plugin) error {
-	for _, pl := range plugins {
+//Run the plugins
+func (loader *PluginLoader) Run() error {
```

Comment:
It is good practice to keep the implementation of given structure in one package.
In this case, it means keeping those functions in one file with the definition of the struct.